File: SP-UI/src/app/models/product.ts
export interface Product {
    id: number
    name: string
    image: string
    price: number
    discount: number
    priceWithDiscount: number
    description: string
    categoryId: number
    categoryName: string
    departmentId: number
    departmentName: string
}

export interface PageProduct {
    id: number
    name: string
    image: string
    price: number
    discount: number
    priceWithDiscount: number
    description: string
    categoryId: number
    categoryName: string
    departmentId: number
    departmentName: string
}

export interface MenuProduct {
    id: number
    name: string
}

export interface ProductCreationDTO {
    name: string
    image: File
    categoryId: number
    departmentId: number
    price: number
    discount: number
    description: string
}

File: SP-API/SP-API/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Mall_API.Entities
{
    public class Product
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Image { get; set; }
        public decimal Price { get; set; }
        [Range(0, 1)]
        public decimal Discount { get; set; }
        public decimal PriceWithDiscount => Price - (Price * Discount);
        public string Description { get; set; }
        public int Weight { get; set; }
        public decimal Height { get; set; }
        public decimal Width { get; set; }
        public decimal Length { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "You must select a {0} from the list")]
        [Display(Name = "Category")]
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "You must select a {0} from the list")]
        [Display(Name = "Department")]
        public int DepartmentId { get; set; }
        public Department Department { get; set; }
    }
}
